### PR TITLE
feat: setup cosmwasm client

### PR DESCRIPTION
**Pull request by gusin13**
Sets up cosmwasm client used for querying and submitting tx's to wasm enabled chains. 

Used in https://github.com/babylonlabs-io/babylon/pull/57

### CODE REVIEW

File: LICENSE [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+"Business Source License" is a trademark of MariaDB Corporation Ab.
+
+
+-----------------------------------------------------------------------------
+
+Parameters
+
+Licensor:             Babylonchain, Inc.
+
+Licensed Work:        vigilante
```
Comment (SebastianElvis) [inline LICENSE:13]:
```suggestion
Licensed Work:        CosmWasm Client
```

File: LICENSE [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+
+License text copyright (c) 2017 MariaDB Corporation Ab, All Rights Reserved.
+"Business Source License" is a trademark of MariaDB Corporation Ab.
+
+
+-----------------------------------------------------------------------------
+
+Parameters
+
+Licensor:             Babylonchain, Inc.
```
Comment (SebastianElvis) [inline LICENSE:11]: Now it's Babylon Labs Ltd.

File: LICENSE [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+
+Licensed Work:        vigilante
+                      The Licensed Work is (c) 2023 Babylonchain, Inc.
```
Comment (SebastianElvis) [inline LICENSE:14]:
```suggestion
                      The Licensed Work is (c) 2024 Babylon Labs, Ltd.
```

**Review by SebastianElvis — APPROVED** [review]

**Comment by SebastianElvis** [comment]
> Used in https://github.com/babylonlabs-io/babylon/pull/57

Also this is used in https://github.com/babylonlabs-io/finality-provider/tree/base/consumer-chain-support/cosmwasmclient and we can replace it with this repo

**Comment by maurolacy** [comment]
If I understand correctly, this is a Go-based cli to replace the previous `cosmwasmclient` module, right?

Also, if I understand correctly, this would be an alternative to use the consumer chain binary (`bcd`) from shell scripts and / or execution calls from Go, right?

**Review by maurolacy — APPROVED** [review]

**Review by RafilxTenfen — APPROVED** [review]
:100:

**Review by Lazar955 — APPROVED** [review]

**Pull request merged**